impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultJavaToolchainFactory.java

```diff
@@ -109,30 +110,35 @@ static class DefaultJavaToolchain implements JavaToolchain {
         }
 
         @Override
+        @Nonnull
         public String getJavaHome() {
             return javaHome;
         }
 
         @Override
+        @Nonnull
         public String getType() {
             return "jdk";
         }
 
         @Override
+        @Nonnull
         public ToolchainModel getModel() {
```

Comment:
but here not. how can this be null please?

<img width="1810" alt="image" src="https://github.com/user-attachments/assets/6525474e-591f-447c-a362-688f327aa920" />

Reply:
https://github.com/apache/maven/pull/2386